src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleAbstractController.php

```diff
@@ -89,6 +92,42 @@ protected function getToolbarButtons()
             ];
         }
 
-        return $toolbarButtons;
+        return array_merge($toolbarButtons, $this->getExtraToolbarButtons());
+    }
+
+    /**
+     * This method will call the actionAdminModuleExtraToolbarButton hook and allow
+     * modules to add some extra buttons to the modules catalogue toolbar.
+     * All the button's specification keys ('href', 'desc', 'icon', 'help') are mandatory
+     */
+    private function getExtraToolbarButtons(): array
+    {
+        try {
+            $extraToolbarContentFromHooks = (new HookFinder())
+                ->setHookName('actionAdminModuleExtraToolbarButton')
+                ->setParams(['controller' => $this])
+                ->find();
+        } catch (CoreException $exception) {
+            return [];
+        }
+
+        $extraToolbarButtons = [];
+
+        // Validation. We check that we have the exact keys
+        foreach ($extraToolbarContentFromHooks as $moduleName => $extraToolbarContentFromHook) {
+            if (!is_array($extraToolbarContentFromHook)) {
+                continue;
+            }
+
+            foreach ($extraToolbarContentFromHook as $buttonIndex => $extraToolbarButton) {
+                if (!empty(array_diff(static::MANDATORY_TOOLBAR_BUTTON_KEYS, array_keys($extraToolbarButton)))) {
```

Comment:
I'm not sure what you want to do. What is the situation you wish to avoid?

Reply:
just want to check that keys given to define the button are exactly the ones expected

Reply:
I don't understand why we need the keys to be exactly the ones expected... although we are trying to add "extra" buttons.

If the buttons added are extra, it could be anything?

Reply:
He is checking the `'href', 'desc', 'icon', 'help'` keys are there, and that there's no other ones, probably because those are the ones that are used in the template

@sowbiba am I right?

Reply:
That's right. And as I said, maybe that check is too strict. I can make it open ;)